src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -823,6 +823,19 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
 
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
+        /// <summary>
+        /// Returns a new ITupleTypeSymbol with the given element types and (optional) element names.
+        /// </summary>
+        /// <remarks>
+        /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// </remarks>
+        public INamedTypeSymbol CreateTupleTypeSymbol(ITypeSymbol[] elementTypes, string[] elementNames = null)
```

Comment:
Anonymous types can't cross methods.  So we never needed to be able to construct symbols for them to put in signatures in a cross language scenario.